internal/future/serving/client/http.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"open-match.dev/open-match/internal/config"
+	"open-match.dev/open-match/internal/future/serving"
+)
+
+// SecureHTTPFromConfig creates an HTTPS client from a configuration.
+func SecureHTTPFromConfig(cfg config.View, prefix string, publicKeyBytes, privateKeyBytes []byte) (*http.Client, string, error) {
```

Comment:
If there's a config file I'd expect paths to the pub/priv key set there and this method loads them in.